docs/upgrade/upgrade-instructions-for-backend-api.md

```diff
@@ -0,0 +1,114 @@
+    - create `app/config/packages/oauth2/.gitignore` as private keys must not be ever pushed
+        ```
+        private.key
+        public.key
+        parameters_oauth.yml
+        ```
+    - create parameters template `app/config/packages/oauth2/parameters_oauth.yml.dist`, this template is used by phing target [backend-api-oauth-keys-generate](/docs/introduction/console-commands-for-application-management-phing-targets.md#backend-api-oauth-keys-generate)
+        ```yaml
+        parameters:
+            oauth2_encryption_key: %%encryption_key%%
+        ```
+    - add default `oauth2_encryption_key` parameter to `app/config/parameters_common.yml`
+        ```diff
+        + oauth2_encryption_key: "0" #placeholder for correctly running composer post script, replaced by app/config/packages/oauth2/parameters_oauth.yml
+        ```
+- add API routes to `app/config/routing.yml`
+    ```diff
+    + shopsys_api_v1:
+    +     resource: "@ShopsysBackendApiBundle/Resources/config/v1/routing.yml"
+    + oauth2:
+    +     prefix: /api
+    +     resource: '@TrikoderOAuth2Bundle/Resources/config/routes.xml'
+    ```
+- add routes for your [custom API](/docs/cookbook/backend-api/creating-custom-api-endpoint.md) to `src/Shopsys/ShopBundle/Resources/config/routing.yml`
+    ```diff
+    + shopsys_shop_api_v1:
+    +     resource: "@ShopsysShopBundle/Controller/Api/V1"
+    +     prefix: /api/v1
+    +     type: annotation
+    ```
+- add security configuration for API routes to `app/config/packages/security.yml`. Be careful, the configuration should be after `administration` and before `frontend`
+    ```diff
+    + backend_api_token:
+    +     pattern: ^/api/token$
+    +     security: false
+    + backend_api:
+    +     pattern: ^/api
+    +     security: true
+    +     stateless: true
+    +     oauth2: true
+    frontend:
+        pattern: ^/
+    ```
+- add following bundles to `app/AppKernel.php`
+    ```diff
+    $bundles = [
+    +     new FOS\RestBundle\FOSRestBundle(),
+    +     new JMS\SerializerBundle\JMSSerializerBundle(),
+    +     new Shopsys\BackendApiBundle\ShopsysBackendApiBundle(),
+    +     new Trikoder\Bundle\OAuth2Bundle\TrikoderOAuth2Bundle(),
+    ```
+    and loading of OAuth parameters
+    ```diff
+    if (file_exists(__DIR__ . '/config/parameters_version.yml')) {
+        $configs[] = __DIR__ . '/config/parameters_version.yml';
+    }
+
+    + if (file_exists(__DIR__ . '/config/packages/oauth2/parameters_oauth.yml')) {
+    +     $configs[] = __DIR__ . '/config/packages/oauth2/parameters_oauth.yml';
+    + }
+    ```
+- run [db-create](/docs/introduction/console-commands-for-application-management-phing-targets.md#db-create) (this one even on production) and `test-db-create` phing targets to install extension for UUID
+- run [backend-api-oauth-keys-generate](/docs/introduction/console-commands-for-application-management-phing-targets.md#backend-api-oauth-keys-generate) phing target to generate OAuth keys
```

Comment:
why do you recommend to run this phing target if they have not set up any API users?

Reply:
You are right, I have removed this step and added different one to read `Introduction to Backend API` so users learn how to start using API in their projects